hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -262,13 +251,26 @@ private static void processRollbackMetadata(HoodieRollbackMetadata rollbackMetad
         partitionToDeletedFiles.get(partition).addAll(deletedFiles);
       }
 
-      if (!pm.getAppendFiles().isEmpty()) {
+      if (pm.getRollbackLogFiles() != null && !pm.getRollbackLogFiles().isEmpty()) {
         if (!partitionToAppendedFiles.containsKey(partition)) {
           partitionToAppendedFiles.put(partition, new HashMap<>());
         }
 
         // Extract appended file name from the absolute paths saved in getAppendFiles()
-        pm.getAppendFiles().forEach((path, size) -> {
+        pm.getRollbackLogFiles().forEach((path, size) -> {
+          partitionToAppendedFiles.get(partition).merge(new Path(path).getName(), size, (oldSize, newSizeCopy) -> {
+            return size + oldSize;
```

Comment:
this size needs to be updated right? is it correct to add this?

Reply:
this is merging entries within rollback and delete files only. Ideally we shouldn't see any merge in this section but only in line 286.

Reply:
still, should we pick the higher value always?